### DISCUSSION TITLE
Is it possible to use an intermediate layer instead of the model directly?

### DISCUSSION THREAD

**Discussion by eduardoarandah** [filamentphp/filament#2492] [Help]
I'm new to the package. 

Looks like filament reads and writes directly to the model. 

Is it possible to read/write to an intermediate layer? 

What if I want a form that triggers a more complex functionality besides crud?

**Comment by danharrin** [comment]
Resources read/write to the model, but you can also write custom forms that don't. What's the actual use case?

**Reply by eduardoarandah** [reply]
Whats the procedure to write to something different from a model?
 Searched in documentation and didn’t find. 
 
Use case is using the tool for something different than CRUD.  

At the beginning, crud is fine, things are easy. But then inevitably the app needs custom forms with custom validation that trigger custom functions.

**Reply by danharrin** [reply]
Filament is split into 3 projects, an admin panel, a form builder, and a table builder.

The admin panel just uses the form builder and table builder internally to render CRUD resources.

There is a "Pages" documentation section that explains how to make your own custom admin panel pages. These pages have the form builder installed by default, so you can build whichever custom forms you wish.

Since pages are livewire components, you have complete control over everything. Nothing is tied to CRUD, nothing is tied to a model. Everything can be changed.

If you don't need CRUD, dont use resources 🤷‍♂️ just check out the form builder on custom pages
